packages/application-shell/src/components/navbar/menu-items.styles.ts

```diff
@@ -110,14 +112,227 @@ const NavlinkClickableContent = styled.div`
   width: 100%;
 `;
 
+const listStyles = css`
+  margin: 0;
+  padding: 0;
+  display: flex;
+  flex-direction: column;
+  min-height: 0;
+  flex: 1 1 0;
+`;
+
+const sublistStyles = css`
+  padding: var(--spacing-30);
+  font-weight: var(--font-weight-for-navbar-link);
+  font-size: var(--font-size-for-navbar-link);
+  background-color: var(--background-color-for-navbar);
+  left: ${NAVBAR.sublistIndentationWhenCollapsed};
+  z-index: -1;
+  list-style: none;
+  position: fixed;
+  display: none;
+`;
+
+const MenuList = styled.ul<
+  MenuGroupProps & {
+    isSublistActiveWhileIsMenuExpanded: boolean;
+    isSublistActiveWhileIsMenuCollapsed: boolean;
+    isSublistCollapsedAndActive: boolean;
+    isSublistCollapsedAndActiveAndAbove: boolean;
+  }
+>`
+  ${(props) => [
+    props.level === 1 && listStyles,
+    getSubmenuPositionBasedOnMenuItemPosition(
+      props.isSubmenuAboveMenuItem,
+      props.submenuVerticalPosition
+    ),
+    props.level === 2 && sublistStyles,
+    // prevent glitchy behavior during the initial render when the submenu's vertical position is evaluated as 0
+    props.submenuVerticalPosition === 0 &&
+      css`
+        visibility: hidden;
+      `,
+    (props.isSublistActiveWhileIsMenuExpanded ||
+      props.isSublistCollapsedAndActive ||
+      props.isSublistCollapsedAndActiveAndAbove) &&
+      css`
+        opacity: 1;
+        display: none;
+        text-align: left;
+        background-color: var(--background-color-for-navbar-when-active);
+
+        /* This pseudo-element is required to enable smooth coursor movement from the main menu item to submenu items with the gap in between */
+        ::before {
+          content: '';
+          position: absolute;
+          display: block;
+          width: calc(${NAVBAR.sublistWidth} + var(--spacing-20));
+          height: ${NAVBAR.itemSize};
+          left: calc(-1 * var(--spacing-20));
+
+          ${getContainerPositionBasedOnMenuItemPosition(
+            props.isSubmenuAboveMenuItem,
+            props.isSublistActiveWhileIsMenuExpanded,
+            props.isSublistActiveWhileIsMenuCollapsed
+          )}
+        }
+      `,
+  ]}
+
+  .highlighted,
```

Comment:
used [here](https://github.com/commercetools/merchant-center-application-kit/blob/a6ea9671704ccb3e40690b8df2c7ef7046985267/packages/application-shell/src/components/navbar/menu-items.tsx#L306)

Reply:
It took me a while to understand this one so I leave a comment with some context for next reviewers.

The `MenuItems` component internally uses `NavLink` one to render the final menu links. The latter is a component coming from `react-router` and it allows for a class name to be provided in order to style the component.
Since the `MenuItems` component is its _parent_, it makes sense to define the class in this file.

I'm wondering if we could use a more specific selector to better indicate the relationship between both components.
Maybe something like this:
```
& .highlighted,
& .highlighted ${Title} {
```